foxpuppet/browser/window.py

```diff
@@ -16,29 +18,42 @@ class BrowserWindow(object):
 
     def __init__(self, selenium, *args, **kwargs):
         self.selenium = selenium
-        self._windows = Windows(selenium)
 
     @property
     def is_private(self):
         """Returns True if this is a Private Browsing window."""
 
-        self.selenium.set_context('chrome')
-        return self.selenium.execute_script("""
+        with self.selenium.context('chrome'):
+            return self.selenium.execute_script(
+                """
                 Components.utils.import("resource://gre/modules/PrivateBrowsingUtils.jsm");
 
                 let chromeWindow = arguments[0].ownerDocument.defaultView;
                 return PrivateBrowsingUtils.isWindowPrivate(chromeWindow);
-            """, self._windows.window_element)
-        self.selenium.set_context('content')
+                """, self.window_element)
 
     def open_window(self, private=False):
-        self.selenium.set_context('chrome')
-        self.selenium.find_element(*self._file_menu_button_locator).click()
-        with self._windows.wait_for_new_window():
+        """Opens a new browser window
+
+        :param private: Optional parameter to open a private browsing window.
+                        Defaults to false.
+        """
+
+        with self.selenium.context('chrome'):
+            handles_before = self.selenium.window_handles
```

Comment:
This is most likely not doing the right thiing because window_handles != chrome_window_handles. Does Geckodriver pass through that property already?